appium/lib/screen-object/accessors/element.rb

```diff
@@ -81,26 +96,116 @@ def dynamic_text_exists? dynamic_text
         end
       end
 
-      def scroll
-        $driver.execute_script 'mobile: scrollTo',:element => element.ref
-        # $driver.execute_script("mobile: scroll",:direction => direction.downcase, :element => element.ref)
+      def gesture(args)
+        Appium::TouchAction.new($driver).swipe(start_x: args[0], start_y: args[1], end_x: args[2], end_y: args[3], duration: args[4]).perform
+      rescue RuntimeError => e
+        raise("Error during gesture in element.rb\n Error Details: #{e}")
       end
 
-      def scroll_to_text(text)
-        $driver.scroll_to(text)
+      def scroll(direction = :down, duration = 1000)
```

Comment:
Remove this `scroll` method and use the on in `ScreenObject::Accessors` to keep it dry.